### PR TITLE
Overlapping ships0

### PR DESCRIPTION
**Pull request by ttakahashi1591**
This pull request is for our Board class. I have added functionality and testing for the following methods:

- `overlapping_ships?`

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -74,24 +98,14 @@
...
       @board.place(@cruiser, ["A1", "A2", "A3"])
-      cell_1 = @board.cells["A1"]
-      cell_2 = @board.cells["A2"]
-      cell_3 = @board.cells["A3"]
-
-      expect(cell_1.ship).to eq @cruiser
-      expect(cell_2.ship).to eq @cruiser
-      expect(cell_3.ship).to eq @cruiser 
 
-      expect(cell_3.ship).to eq cell_2.ship
```
Comment (vsanantone) [inline spec/board_spec.rb:94]: The methodical testing procedures  based on the interaction pattern really solidified our accuracy in TDD.

File: lib/board.rb [hunk 2]
```diff
@@ -56,4 +57,8 @@ def place(ship, coordinates)
       # coordinates.each { |coordinate| @cells[coordinate].place_ship(ship)}
     end
   end
+
+  def overlapping_ships?(coordinates)
```
Comment (vsanantone) [inline lib/board.rb:61]: Great example of refactored code from "do end" to "{}".

File: spec/board_spec.rb (hunk 1 above)
Comment (ttakahashi1591) [reply spec/board_spec.rb:94]: Thank you! I noticed that it was important for us to include the entire interaction pattern in order for the test to work.

File: lib/board.rb (hunk 2 above)
Comment (ttakahashi1591) [reply lib/board.rb:61]: Appreciate it! We can keep our code the way it is for now, but this was just an example of how we can make the `place` method into one line.

**Comment by vsanantone** [comment]
Great! I'm going to merge this, and I have left a few comments in the file's changed section that you can look at when you get a chance.

**Pull request merged**